### ISSUE TITLE
Question: Unicable with DiseqC

### ISSUE THREAD

**Issue by lars18th** [catalinii/minisatip#850]
Hi @9000h and others,

I open a new issue to ask about a question related to an special environment mixing a Unicable LNB and one DiseqC 4x1 swith. I know they're other threads discussing Unicable environments, but I prefer to open another one. Please, comment your opinions/experiences. Thank you!

In my current environment I can use _only_ one wire. This can't be changed, so I want to achieve the maximum with this limitation. And the wire has the lower part occupied with the terrestrial band. So, full band operation is out-of-scope (neither with stacking). In this scenario, for receiving 4 satellite positions (as I have now) using Unicable the only solution is to use programable devices. However, the cost of this is high. Then I'm thinking on a poor-man's solution. But first I want to know your opinion about this idea.

The current infrastructure is the typical 4x1 DiseqC 1.0 swith + the terrestrial signal. This works without troubles... until you want to use more than one DVB-S2 tuners. In this case, the solution is to configure a master-slave mode. Then the slave tuners can use only the master band, if you can solve the power problem. But then you're limited to one of the 16 potential subbands (4 bands for each of the 4 sat positions). So the idea is to provide a more flexible solution _with a low cost_.

So, my current idea is to mix the current 4x1 DiseqC with one Unicable LNB. See this diagram:
```
     LNB 1    LNB 2    LNB 3    LNB 4
   (unicable)   |        |        |
       |        |        |        |
     --------------------------------
     |        4x1 DiseqC            |
     --------------------------------
                   |
                   |           |----- Tuner 1
                   ------------|----- Tuner 2
                               |----- Tuner 3
```
The objective here is this:
- When using the sat position A with the Unicable LNB all tuners can use the 4 bands. So if this is your preferable sat position, then you'll have access to all services.
- But when you want to use other satellite positions, then in this case only the selected master band can be accessed.

If you compare this with the regular 4x1 DiseqC configuration then you win the option to access to the entire satellite position A for all tuners, maintaining the same limitations for the rest of satellites. But the cost of this is the price of a new inexpensive SCR LNB to replace the LNB 1. And nothing more.

However, the question is this: If I use minisatip for the master tuner... then how I need to configure it for this? The objective is to send FIRST the DiseqC 1.0 selection command, and after the switch has selected the route to the LNB 1 then send the Unicable command. Futhermore, it requires to support to not use the Unicable protocol but the regular tunning for the LNBs 2-4.

You think this will work? Or you have other proposals? And with regard to tuners, they will all be driven by minisatip.

**Comment by catalinii** [comment]
Hey @lars18th,

I am actually interested in this feature as well, but do not have any unicable lnb.

I think the first question is, does it work?

If you have the setup already made, sending diseqc command then the unicable should show if it works.

If it does, we need to figure out how do we design it.

**Comment by 9000h** [comment]
may there are some related info in the satip-axe git to find

**Comment by lars18th** [comment]
Hi @catalinii ,

> Hey @lars18th,
> I am actually interested in this feature as well, but do not have any unicable lnb.

Great! I hope we can check & support this. If others (for example @9000h) think that this can work then I'll purchase one cheap unicable lnb to test it.

> may there are some related info in the satip-axe git to find

Please @9000h , can you share your opinion?

**Comment by catalinii** [comment]
What about using 4 unicable each using different pin? That way minisatip can decide which lnb to control by using the appropriate pin.

**Comment by 9000h** [comment]
an cheap and simple solution would be
![598d6d31037b1-1502440753](https://user-images.githubusercontent.com/3009073/120082078-4132c700-c0c1-11eb-8dd7-07da2117bbf4.png)

**Comment by lars18th** [comment]
Hi @9000h ,

> an cheap and simple solution would be

Sorry, but this is the **EXPENSIVE** solution. So it's completely out of context. One Unicable/SRC LNB is around 2x or 3x the price of a regular LNB. The price of such units is: 4 x Widebands LNBs about 3x the price of a regular LNB + 2 programble unicable switches of a cost of 15x the price of a regular LNB. So a total of 12+15+15=42 times the price of a regular LNB. While chaging one LNB with my suggestion is only 2 or 3 times the price of a regular LNB. Therefore we need a more cheap solution.

**Comment by lars18th** [comment]
Hi @catalinii ,

> What about using 4 unicable lnb each using different pin? That way minisatip can decide which lnb to control by using the appropriate pin.

The problem using more than one unicable LNB in the same wire is this: When you make a request then all LNBs do a response injecting signal in this frequency, so you'll see the data from the different satellites overlaped. And obviously this doesn't work. The problem is that manufacturers are programming their scr LNBs to response to positions A+B+C+D. So it will be preferable if the LNB will have some switch to select: A or B or C or D (and A+B+C+D if you want). But this is not the case. And programble LNBs are expensive (between 3-4 times the price of a cheap unicable LNB).

And a solution of mixing Unicable and DiseqC will not work with more than one Unicable because you then need to share the same satellite position for all tuners. And this then not solves the problem. Therefore my suggestion is to almost for the primary position you can use the four bands. Then, please can you add some code to enforce a DiseqC command _before_ sending the SCR command?

**Comment by catalinii** [comment]
My point was that if you send the command using pin 1234 which is the pin configured for LNB1 (and all other LNBs have other pins) only LNB1 will actually perform the action.

AsI do not have unicable lnbs I cannot test this.

In the current configuration is either unicable or diseqc and not both. That's I am asking us to test if it works.

**Comment by catalinii** [comment]
If the tuning sequence is:
- choose a free unicable slot X and send it to LNB y using the correct pin
- when the user stops the channel tune LNB y to slot 0
In this way any LNB can use up to 31 slots if other LNBs are not in use which is better than assigning a static number of slots to each LNB.

If this works it will be better because you can do dynamic assignment of slots based on requirements.

**Comment by lars18th** [comment]
Hi @catalinii ,

When using multiple unicable LNBs, please note this:

- Not all (in fact only few) are programable. This is the main problem.
- Regular unicable LNBs are not using PIN protection. So it's difficult that two LNBs with the same slot frequencies are using different PIN values.

However, I see your idea: If two unicable LNBs are using the same 24 or 32 ports, and they use PINs and the values are different, then you can use the PIN to send commands to an specific LNB. That's a good idea. Then you can send first the command to disable the port in one LNB and then enable in another, in case you want to change to another sat position. Futhermore, if you use only some ports for one satellite and other ports for other satellite, then you can overcome this two-command behaviour, as you can send commands only to one LNB.

If this is your idea, then I'll search for some (low cost) unicable LNBs with different PINs. So, if we found them, then you'll want to implement support for this environment?

**Comment by catalinii** [comment]
This requires minimum configuration to work. The static part should work with 1 adapter to slot implementation.

For the dynamic mode the source should be extended to cover this (ability to select satellite by any adapter using src=)

**Comment by lars18th** [comment]
Hi @catalinii ,

I don't find low cost LNBs unicable with different user bands AND pins. If you found any, please share with us.

**Comment by 9000h** [comment]
https://www.amazon.de/dp/B01B4ZHZBG

**Comment by Neolo** [comment]
Right now even last version of minisatip **does not work** with Unicable 2 in **dynamic** mode of user bands!  https://github.com/catalinii/minisatip/issues/840

**Comment by lars18th** [comment]
Hi @Neolo ,

Thank you for commenting it. Then until minisatip will work with the dynamic mode I'll not purchase anything. In fact, the static mode is compatible with _any_ device… as you only need to tune to the central frequency (so nothing is involved).

**Comment by Neolo** [comment]
> Hi @Neolo ,
> 
> Thank you for commenting it. Then until minisatip will work with the dynamic mode I'll not purchase anything. In fact, the static mode is compatible with _any_ device… as you only need to tune to the central frequency (so nothing is involved).

Yes, unfortunately, I've compared with other software and external tuners with mysetup, and in all tests only minisatip fails to tune the transponder in dyn mode. As a workaround I tune needed transponder using Astra Cesbo on central freq and then I start minisatip on it. Ugly but works.

**Comment by Delitants** [comment]
This problem has never been solved for me, I have confirmed with multiple tests that combination of Inverto DLU-UWT110-CU010-32P multiswitch (Unicable 2 **dynamic** mode) and tuner TBS 6590 - **does not work** for minisatip to tune a transponder.
Moreover, minisatip is not even able to open TBS 6590 unless I execute a hack:
mv /dev/dvb/adapter1/frontend0 /dev/dvb/adapter1/frontend2
mv /dev/dvb/adapter1/frontend1 /dev/dvb/adapter1/frontend0
mv /dev/dvb/adapter1/frontend2 /dev/dvb/adapter1/frontend1
